CONTRIBUTING.md

```diff
@@ -0,0 +1,142 @@
+# Contributing guidelines
+
+## Formatting
+
+By default, formating is checked by gofmt tool, where almost format all the
+cases.
+
+The project does not have a line length limit, but the team try to be as small
+as possible, keeping the code clean on reading. Having an 80-120 char limit is
+an excellent number to split the line.
+
+## Style
+
+The project follows the Golang project guidelines that you can find in the
+following link:
+
+[https://github.com/golang/go/wiki/CodeReviewComments]()
+
+## Developer environment
+
+By default, all k4e projects rely on Makefiles, where users can run all the
+workflows operations.
+
+To make it easier, all Makefiles implement a help section to see the actions you
+can run. Here is an example:
+
+```
+--> make help
+Usage:
+  make <target>
+
+General
+  help             Display this help.
+
+Development
+  manifests        Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
+  generate         Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
```

Comment:
For development there is another target named `gen-manifests `

It is defined as:
```
$(eval TMP_ODIR := $(shell mktemp -d))
gen-manifests: manifests kustomize # generates manifests for deploying the operator into k4e-operator.yaml
        cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
        $(KUSTOMIZE) build config/default > $(TMP_ODIR)/k4e-operator.yaml
```

It is similar to `make deploy` without deploying it on the target cluster. Instead, the generated operator installation yaml is stored to a local file.
It is useful since it can be use to install the operator on any cluster, at any time without a need to have the project cloned or rebuilt.

```
→ IMG=quay.io/masayag/k4e-operator make gen-manifests 
/home/masayag/work/go/src/github.com/jakub-dzon/k4e-operator/bin/controller-gen "crd:trivialVersions=true,preserveUnknownFields=false" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
cd config/manager && /home/masayag/work/go/src/github.com/jakub-dzon/k4e-operator/bin/kustomize edit set image controller=quay.io/masayag/k4e-operator
/home/masayag/work/go/src/github.com/jakub-dzon/k4e-operator/bin/kustomize build config/default > /tmp/tmp.VXeIHv0BZG/k4e-operator.yaml
```

`/tmp/tmp.VXeIHv0BZG/k4e-operator.yaml` can then be used to install the operator.
It is used by the `release` target, but I tend to use it locally when working on remote clusters.
I'm okay with omitting `gen-manifests` and `release` since we don't have any plans to release it, but its there.